apps/custom-config/app.js

```diff
@@ -2,266 +2,86 @@ var _ = require('@yoda/util')._
-        action: action,
-        success: setStatus
-      }])
-    }
-    activity.wormhole.sendToApp(VT_WORDS_TOPIC, sendObj)
-  }
+module.exports = function CustomConfig (activityIn) {
+  activity = activityIn
+  activity.on('ready', onReady)
+  activity.on('request', onRequest)
+  activity.on('url', onUrl)
+}
 
-  function onPickupSwitchStatusChanged (action, isFirstLoad) {
-    if (action) {
-      property.set('sys.pickupswitch', action, 'persist')
-      if (!isFirstLoad) {
-        if (action === SWITCH_OPEN) {
-          activity.tts.speak(PICKUP_SWITCH_OPEN).then(() => activity.exit())
-        } else if (action === SWITCH_CLOSE) {
-          activity.tts.speak(PICKUP_SWITCH_CLOSE).then(() => activity.exit())
-        } else {
-          activity.tts.speak(CONFIG_FAILED).then(() => activity.exit())
+/**
+ * skill url was requested
+ * @param url - skill url
+ */
+function onUrl (url) {
+  var urlObj = Url.parse(url)
+  var queryObj = urlObj.query
+  var path = urlObj.pathname.substr(1)
+  logger.info(`on Url---->is called ${path}  ${JSON.stringify(urlObj.query)}`)
+  if (path === 'firstLoad') {
+    var customConfig = safeParse(queryObj.config)
+    if (customConfig && typeof customConfig === 'object') {
+      for (var field in customConfig) {
+        if (customConfig.hasOwnProperty(field) && urlMap.hasOwnProperty(field)) {
+          urlMap[field](safeParse(customConfig[field]))
         }
       }
     }
-  }
-
-  function onStandbyLightSwitchStatusChanged (action, isFirstLoad) {
-    if (action) {
-      property.set('sys.standbylightswitch', action, 'persist')
-      if (action === SWITCH_OPEN) {
-        activity.light.play(STANDBY_LIGHT_JS, {}, {shouldResume: true})
-      } else if (action === SWITCH_CLOSE) {
-        activity.light.stop(STANDBY_LIGHT_JS)
-      }
-      if (!isFirstLoad) {
-        if (action === SWITCH_OPEN) {
-          activity.tts.speak(LIGHT_SWITCH_OPEN).then(() => activity.exit())
-        } else if (action === SWITCH_CLOSE) {
-          activity.tts.speak(LIGHT_SWITCH_CLOSE).then(() => activity.exit())
-        } else {
-          activity.tts.speak(CONFIG_FAILED).then(() => activity.exit())
-        }
-      }
+  } else {
+    var func = urlMap[path]
+    if (func) {
+      func(queryObj)
+    } else {
+      logger.warn(`skill path [${path}] is not hit`)
     }
   }
+}
 
-  function onWakeupSwitchStatusChanged (action, isFirstLoad) {
-    if (action) {
-      property.set('sys.awakeswitch', action, 'persist')
-      if (!isFirstLoad) {
-        if (action === SWITCH_OPEN) {
-          activity.tts.speak(WAKE_SOUND_OPEN).then(() => activity.exit())
-        } else if (action === SWITCH_CLOSE) {
-          activity.tts.speak(WAKE_SOUND_CLOSE).then(() => activity.exit())
-        } else {
-          activity.tts.speak(CONFIG_FAILED).then(() => activity.exit())
-        }
-      }
+/**
+ * activity is ready
+ */
+function onReady () {
+  activity.get().then(config => {
+    cloudgw = new CloudGW(config)
```

Comment:
As we have decided in #110, the activity.get might be deprecated, so I think we should not call this any more in our new API, suggest we provide a new descriptor `CloudGw` that could be used.

Reply:
/cc @legendecas

Reply:
It's not implemented yet. We could do a glob replacement after the new API Implementation.